aten/src/ATen/ExpandUtils.h

```diff
@@ -446,45 +474,67 @@ static inline Tensor sum_to(
   if (shape.size() == 0) {
     return tensor.sum();
   }
-  c10::SmallVector<int64_t, 8> reduce_dims;
+
   const at::IntArrayRef sizes = tensor.sizes();
-  const int64_t leading_dims = sizes.size() - shape.size();
-  for (const auto i : c10::irange(leading_dims)) {
-    reduce_dims.push_back(i);
-  }
-  for (int64_t i = leading_dims; i < static_cast<int64_t>(sizes.size()); ++i) {
-    if (shape[i - leading_dims] == 1 && sizes[i] != 1) {
-      reduce_dims.push_back(i);
-    }
-  }
+  auto reduce_dims = compute_reduce_dims<int64_t, IntArrayRef>(sizes, shape);
   if (!reduce_dims.empty()) {
     tensor = tensor.sum(reduce_dims, /*keepdim=*/true);
   }
-  if (always_return_non_view) {
-    // This is only actually used by the functionalization pass.
-    // We want to be able to guarantee that this function doesn't return a view
-    // of the input.
-    return leading_dims > 0 ? at::view_copy(tensor, shape) : tensor.clone();
-  } else {
-    return leading_dims > 0 ? tensor.view(shape) : tensor;
+
+  auto sym_shape = c10::SymIntArrayRef(
+      reinterpret_cast<const c10::SymInt*>(shape.data()), shape.size());
+
+  const int64_t leading_dims = sizes.size() - shape.size();
+  return functionalize_sum(
+      leading_dims, tensor, sym_shape, always_return_non_view);
+}
+
+// Sums `tensor` repeatedly to produce a tensor of shape `shape`.
+// Precondition: is_expandable_to(shape, tensor.sizes()) must be true
+static inline Tensor sum_to(
+    Tensor tensor,
+    const c10::SymIntArrayRef shape,
+    bool always_return_non_view = false) {
+  if (shape.size() == 0) {
+    return tensor.sum();
   }
+  auto sym_sizes = tensor.sym_sizes();
+  auto reduce_dims =
+      compute_reduce_dims<c10::SymInt, c10::SymIntArrayRef>(sym_sizes, shape);
+  if (!reduce_dims.empty()) {
+    tensor = tensor.sum_symint(reduce_dims, /*keepdim=*/true);
+  }
+
+  auto leading_dims = sym_sizes.size() - shape.size();
+  return functionalize_sum(leading_dims, tensor, shape, always_return_non_view);
```

Comment:
Is there any reason to dupe this? It seems to me like we should only have the SymInt version of this function.